geyser-plugin-interface/src/geyser_plugin_interface.rs

```diff
@@ -438,4 +493,15 @@ pub trait GeyserPlugin: Any + Send + Sync + std::fmt::Debug {
     fn entry_notifications_enabled(&self) -> bool {
         false
     }
+
+    /// Check if the plugin is interested in accounts data before
+    /// exection of transaction
+    /// Default is false -- if plugin is interested in accounts data
+    /// before execution, return true
+    /// This feature will require more memory and could lead
+    /// in degraded performance as the pre transaction execution states
+    /// will be saved in memory, so activate only if required
+    fn enable_pre_trasaction_execution_accounts_data(&self) -> bool {
+        false
+    }
 }
```

Comment:
If we can make this fast and light enough to have it always on, then let's. It would be preferable if `previous_account_state` didn't have to be an `Option` and if we didn't have to make developers think about which knobs to turn on/off.

Did you measure the memory impact of sending the prior account state in the notification? I think we need to take some measurements with this feature present and absent as you slam the db with `store_*()`. Something [like this](https://github.com/anza-xyz/agave/blob/e114e3bc4f12d6da687344dc8eb85e36738ef9c5/accounts-db/benches/accounts.rs#L36-L77), but with a non-`None` notifier. You could also write a [bench suite like this](https://github.com/anza-xyz/agave/blob/e114e3bc4f12d6da687344dc8eb85e36738ef9c5/accounts-cluster-bench/src/main.rs#L343-L361) that does a self-transfer or something, to really test it from the outside in.